docs/sat-solver.md

```diff
@@ -0,0 +1,383 @@
+
+<img src="https://render.githubusercontent.com/render/math?math=x_{j} \in \{0,1\}, a_{ij},b_{i}, c_{j} \in N_{0}^{+}, i \in M">
+
+<img src="https://render.githubusercontent.com/render/math?math=M = 1, \dots, m">
+
+Where:
+
+- Each <img src="https://render.githubusercontent.com/render/math?math=c_{j}">
+  is a non-negative integer cost, associated with variable
+  <img src="https://render.githubusercontent.com/render/math?math=x_{j}, j \in N">.
+- <img src="https://render.githubusercontent.com/render/math?math=a_{ij}">
+  denotes the coefficients of the literals
+  <img src="https://render.githubusercontent.com/render/math?math=l_{j}">
+  in the set m of linear constraints (a literal being a Boolean variable).
+
+In our case, a package _x_ belongs to the repository _P_, where _P_ is the set
+of tuples
+<img src="https://render.githubusercontent.com/render/math?math=P = \{ (repoUrl, name, semverRange, ns), \dots \}">.
+
+## Constraints definitions
+
+We will define Pseudo-Boolean functions that will be optimized by selecting
+their input values. This Pseudo-Boolean functions can then be implemented in
+Hypper easily.
+
+### Installation
+
+<img src="https://render.githubusercontent.com/render/math?math=x_{1}"> is installed:
+
+<img src="https://render.githubusercontent.com/render/math?math=x_{1} \geq 1">,
+with <img src="https://render.githubusercontent.com/render/math?math=x_{1} \in P">
+
+For simplicity, the tuple of package
+<img src="https://render.githubusercontent.com/render/math?math=x = (repoUrl, name, semverRange = 1.0, ns)">
+is represented as
+<img src="https://render.githubusercontent.com/render/math?math=x_{1}">.
+
+
+### Depends
+
+<img src="https://render.githubusercontent.com/render/math?math=p_{1}"> depends on
+<img src="https://render.githubusercontent.com/render/math?math=x_{1}">:
+
+<img src="https://render.githubusercontent.com/render/math?math=x_{1} - p_{1} \geq 0">
+
+Installing <img src="https://render.githubusercontent.com/render/math?math=p_{1}"> implies
+installing <img src="https://render.githubusercontent.com/render/math?math=x_{1}">.
+Although <img src="https://render.githubusercontent.com/render/math?math=x_{1}"> may be
+installed without <img src="https://render.githubusercontent.com/render/math?math=p_{1}">.
+
+
+### Depends on multiple versions
+
+If <img src="https://render.githubusercontent.com/render/math?math=p_{1}">
+depends on multiple versions of _x_, for example,
+<img src="https://render.githubusercontent.com/render/math?math=x_{1}"> and
+<img src="https://render.githubusercontent.com/render/math?math=x_{2}">:
+
+<img src="https://render.githubusercontent.com/render/math?math=x_{1} %2B x_{2} - p_{1} \geq 0">
+
+
+### Conflicts
+
+If <img src="https://render.githubusercontent.com/render/math?math=y_{2}"> conflicts with
+<img src="https://render.githubusercontent.com/render/math?math=x_{1}">:
+
+<img src="https://render.githubusercontent.com/render/math?math=x_{1} %2B y_{3} \leq 1">
```

Comment:
Should be `y_{2}` I guess

Reply:
Indeed! thanks :).

Reply:
fixed in https://github.com/rancher-sandbox/hypper/pull/171/commits/8de17228510efc1f06b80190e5b7dd4d6fa0b0db.